crates/cli/src/subcommands/sql.rs

```diff
@@ -72,50 +82,205 @@ pub(crate) async fn parse_req(mut config: Config, args: &ArgMatches) -> Result<C
+    ///  0xaba52919637a60eb336e76eed40843653bfb3d9d94881d78ab13dda56363b993 | (none = ()) | true
+    /// ```
+    async fn render_tabled(self, mut out: impl AsyncWrite + Unpin) -> anyhow::Result<()> {
+        // Print only `OK for empty tables as it's likely a command like `INSERT`.
+        if self.stmt_results.is_empty() {
+            if self.opts.with_timing {
+                out.write_all(self.fmt_timing().as_bytes()).await?;
+            }
+            out.write_all(b"OK").await?;
+            return Ok(());
         }
-        println!("OK");
-        return Ok(());
-    };
 
-    stmt_result_json
-        .iter()
-        .map(|stmt_result| {
+        for (stmt_result, sep) in self.stmt_results.iter().zip(self.sep_by(b"\n\n")) {
             let mut table = stmt_result_to_table(stmt_result)?;
-            if with_stats {
-                // The `tabled::count_rows` add the header as a row, so subtract it.
-                let row_count = print_row_count(table.count_rows().wrapping_sub(1));
-                table.with(tabled::settings::panel::Footer::new(row_count));
+            if self.opts.with_row_count {
+                let rows = stmt_result.rows.len();
+                table.with(Footer::new(format!(
+                    "({} {})",
+                    rows,
+                    if rows == 1 { "row" } else { "rows" }
+                )));
+            }
+            out.write_all(table.to_string().as_bytes()).await?;
+            if let Some(sep) = sep {
+                out.write_all(sep).await?;
+            } else {
+                out.write_all(b"\n").await?;
+            }
+        }
+
+        if self.opts.with_timing {
+            out.write_all(self.fmt_timing().as_bytes()).await?;
+        }
+        out.flush().await?;
+        Ok(())
+    }
+
+    /// Renders each query result as a sequence of CSV-formatted lines.
+    ///
+    /// The column names are printed as a comment (a line starting with '#') at
+    /// the beginning of each row.
+    ///
+    /// # Example
+    ///
+    /// A query `SELECT * FROM Message; SELECT * FROM Users;` against the
+    /// `quickstart-chat` application may return the following:
+    ///
+    /// ```text
+    /// # sender,sent,text
+    /// 0xaba52919637a60eb336e76eed40843653bfb3d9d94881d78ab13dda56363b993,1718611618273455,hello
+    /// # identity,name,online
+    /// 0xaba52919637a60eb336e76eed40843653bfb3d9d94881d78ab13dda56363b993,(none = ()),true
+    /// ```
+    async fn render_csv(self, mut out: impl AsyncWrite + Unpin) -> anyhow::Result<()> {
+        for stmt_result in &self.stmt_results {
+            let mut csv = csv_async::AsyncWriter::from_writer(&mut out);
+            let StmtResultJson { schema, rows } = stmt_result;
+            let ts = Typespace::EMPTY.with_type(schema);
+
+            // Render column names as a comment, e.g.
+            // `# sender,sent,text`
+            for (i, name) in schema.names().enumerate() {
```

Comment:
It's called "spacetime-flavored csv" :stuck_out_tongue: 

I don't know what else to do, because we don't have a named table here (unlike the subscribe command). Maybe you have an idea?

Reply:
Is not clear to me why put the names as comments.

Reply:
Because you get variable-width rows if you run a multi-statement query.

Reply:
I don't follow, how this help? A simple example?

Reply:
I gave an example in the commentary, what’s unclear about it?

Reply:
After discussing it, I understand now the reason, is for separate the outputs of different queries.

In psql:
```bash
❯ psql -U postgres  -d "Name" --csv  -c "SELECT name from company; SELECT name, username from company"
name
apple   | admin
name, username
google   | jhon
```

I suggest using `#` or `\n` to mark where is the output of the next query:


```csv
name
apple   | admin
#
name, username
google   | jhon
```

or

```csv
name
apple   | admin

name, username
google   | jhon
```

Or add a parameter so the user selects what to use for this.